mailpoet/lib/Segments/DynamicSegments/Filters/DateFilterHelper.php

```diff
@@ -16,14 +16,14 @@ class DateFilterHelper {
   const IN_THE_LAST = 'inTheLast';
   const NOT_IN_THE_LAST = 'notInTheLast';
 
-  public function getValidOperators(): array {
+  public static function getValidOperators(): array {
```

Comment:
Can I ask why did you change the non-static public method to static?
I'm asking because `DateFilterHelper` is a service that should be injected via DI, and in my opinion, those services shouldn't contain static methods.

Reply:
Thank you for bringing this up. In the back of my mind I've been thinking about updating a lot of the validation code in `FilterDataMapper` to rely on the filter classes themselves, because we end up having to perform most of the same checks in both places. I imagined the constructor growing very large if we needed to inject every single filter that performs any validation, which is why the static method seemed nicer to me.

That being said, I totally agree with you and I will change this back to a public method and inject it into FilterDataMapper.

Reply:
I updated the code in 823fdb331d31a7d6aec7635e144fdd7a14cd61ac

For the constructor of `FilterDataMapper`, would it be better to inject the container wrapper instead, or is it ok how I did it?

Reply:
@johnolek I guess we should avoid using `ContainerWrapper::getInstance()` when it's not necessary or is a good reason for that. Do you think that there is a reason for it? I remember we use it sometimes in temporary code, especially in places that should be removed after finishing Doctrine refactoring.

Reply:
No good reason. I’ve seen it in enough places that I thought it might be idiomatic in some cases. I’ll update this to use injection as well.